packages/block-editor/src/store/selectors.js

```diff
@@ -2968,43 +2976,66 @@ export const getBlockEditingMode = createRegistrySelector(
 				return 'disabled';
 			}
 
+			if ( editorMode === 'navigation' ) {
+				const sectionRootClientId = getSectionRootClientId( state );
+
+				// The root section is "default mode"
+				if ( clientId === sectionRootClientId ) {
+					return 'default';
+				}
+
+				// Sections should always be contentOnly in navigation mode.
+				const sectionsClientIds = getBlockOrder(
+					state,
+					sectionRootClientId
+				);
+				if ( sectionsClientIds.includes( clientId ) ) {
+					return 'contentOnly';
+				}
+
+				// Blocks outside sections should be disabled.
+				const isWithinSectionRoot = isWithinBlock(
+					state,
+					clientId,
+					sectionRootClientId
+				);
```

Comment:
Do we also want to do this in `zoom-out` mode? I'd say we might well.

Aside: this is why maintaining x2 similar modes during the 6.7 release cycle is going to be tricky.

Reply:
What if we just change the condition to `if [ "zoom-out", "navigation" ].includes( editorMode )` (until we remove zoom-out)

Reply:
Here are the two differences that exist with zoom-out today:

 - The root container is "default" in "navigation" but "contentOnly" in zoom-out.
 - Zoom-out doesn't enable "content blocks", everything is disabled within sections. 

Do we want to unify that?

Reply:
Given that currently zoom-out disables everything that is not a section, the bug covered by this PR is actually already covered by "zoom-out" but if we change zoom-out to enable editing "content " blocks, then we'll have the same issue. So for now, it seems safe to move forward.

Reply:
Thank you 🙇‍♂️